### ISSUE TITLE
Provide completions for spaghetto

### ISSUE THREAD

**Issue by deemp** [purescript/spago#977]
I used [purescript-overlay](https://github.com/thomashoneyman/purescript-overlay) and couldn't get completions for `spaghetto 0.93.9` using `nix run .#spago-unstable -- --bash-completion-script`. It failed with this message

```
spago
    Unexpected argument:
        --bash-completion-script
...
```

Then, I searched the `spago` repo and found no `completion` entries in `spaghetto`.

It would be nice to have a documented option for `spaghetto` completions.

**Comment by f-f** [comment]
I think this should be raised [upstream in the argparsing library](https://github.com/natefaubion/purescript-argparse-basic) that we use?

**Comment by deemp** [comment]
You can probably use https://github.com/f-o-a-m/purescript-optparse

**Comment by f-f** [comment]
I am not really up for rewriting the CLI handling myself, but I'd be willing to merge a PR that implements this and doesn't make it substantially worse to maintain

**Comment by deemp** [comment]
Closed by #980
